torch/csrc/distributed/rpc/rref.cpp

```diff
@@ -131,32 +130,39 @@ const ForkId& UserRRef<T>::forkId() const {
 }
 
 template <>
-IValue UserRRef<IValue>::toHere() {
+std::shared_ptr<ivalue::Future> UserRRef<IValue>::toHere() {
+  auto future = std::make_shared<ivalue::Future>(nullptr);
   auto agent = RpcAgent::getDefaultRpcAgent();
-  std::shared_ptr<FutureMessage> fm = agent->send(
+  auto fm = agent->send(
```

Comment:
When the `ScriptRRefFetchCall`/`PythonRRefFetchCall` arrives, the owner might not exist yet in the callee. So, when handling the fetch all, it could be the first time the callee see that `RRefId` and will need to create a new `OwnerRRef`, which requires to know the type. 

Another option is that, instead of blocking on `OwnerRRef::getValue()`, request handler it can block (or add callback) on `RRefContext` until the `OwnerRRef` becomes ready. In that way, we don't need two messages any more. We can add that in a follow up PR.

Reply:
I see. In summary, sometimes users know the types before owners. 
For example, a user asks to run a user function. The return value will be returned as an RRef. The type of the return value is known by the user node first.